wa/workloads/aitutu/uiauto/app/src/main/java/com/arm/wa/uiauto/aitutu/UiAutomation.java

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+*/
+
+
+package com.arm.wa.uiauto.aitutu;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.graphics.Rect;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+import android.support.test.uiautomator.UiScrollable;
+import android.view.KeyEvent;
+import android.util.Log;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    public static String TAG = "UXPERF";
+
+    @Test
+    public void setup() throws Exception {
+        acceptPermissions();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+    }
+
+    @Test
+    public void extractResults() throws Exception {
+        getScores();
+    }
+
+    public void acceptPermissions() throws Exception {
+        UiSelector selector = new UiSelector();
+        UiObject perm1 = mDevice.findObject(selector.textContains("1 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm2 = mDevice.findObject(selector.textContains("2 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm3 = mDevice.findObject(selector.textContains("3 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm4 = mDevice.findObject(selector.textContains("4 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject allow = mDevice.findObject(selector.textContains("ALLOW")
+                                             .className("android.widget.Button"));
+        if (perm1.exists()) {
+            allow.click();
+        }
+        if (perm2.exists()) {
+            allow.click();
+        }
+        if (perm3.exists()) {
+            allow.click();
+        }
+        if (perm4.exists()) {
+            allow.click();
+        }
+
+        UiObject next = mDevice.findObject(selector.textContains("NEXT")
+                                             .className("android.widget.Button"));
+        waitObject(next);
+        next.click();
+
+        UiObject cancel = mDevice.findObject(selector.textContains("CANCEL")
+                                             .className("android.widget.Button"));
+        waitObject(cancel);
+        cancel.click();
+    }
+
+    public void runBenchmark() throws Exception {
+        UiSelector selector = new UiSelector();
+        //Start the tests
+        UiObject start = mDevice.findObject(selector.textContains("Start Testing")
```

Comment:
By default the target cache is cleared resulting in the additional download to have to be repeated for each run. 
Is this necessary? If not `self.clear_data_on_reset` could be set to `False` for the python workload to prevent this, although in that case it should be noted in the workload description that the first run will involve downloading additional assets and therefore will significantly differ to additional runs.

Reply:
It is possible that in future releases the dependencies that the app has to download will change. I'd like to keep this as part of the workload for future proofing.

Reply:
Would the application itself not detect this and re download the relevant assets? 
I was just thinking that if you're measuring anything aside from the workload reported scores, e.g. device power, time etc these would be very much dependent on the speed of the Internet connection at the particular time the workload was ran and would be comparable across iterations?

Reply:
Can WA manage at what point in the workload it records instrumentation? If so that would solve this problem nicely as the code for the actual testing in the run stage would be all we'd want to record against? 

I'm not sure of the behaviour when assets are missing as it hasn't happened yet. I just wanted to future proof and keep things as generic as possible.

Reply:
The idea WA uses is that the `run` section of the workload is only meant to be the work that should be measured and all the downloading / configuration is meant to be performed within the `setup` section, therefore WA will kick off and stop the instrumentation as close as it can before/after that method is executed so unfortunately it does not support configurable starting positions within execution sections.  

Is there any way to split the two sections or wait for the download to complete before ending the `setup` phase and then starting the `run` stage?

Reply:
I could add a wait for the "downloading" text to disappear and then start the run stage, the app would still be performing actions though unless there is a way to force pause it?

Reply:
Hmm... I'm not aware of anyway to forcibly pause an application so waiting for the download to complete might be the better option.. Having a look at the application it does seem to have a intermediate stage checking the resources and atleast the delay / offset should be consistent between different runs?